mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -717,6 +719,49 @@ class SparseVector(
       new SparseVector(size, ii, vv)
     }
   }
+
+  override def argmax: Int = {
+    if (size == 0) {
+      -1
+    } else {
+
+      var maxIdx = 0
+      var maxValue = if(indices(0) != 0) 0.0 else values(0)
+
+      foreachActive { (i, v) =>
+        if (v > maxValue) {
+          maxIdx = i
+          maxValue = v
+        }
+      }
+
+      // look for inactive values incase all active node values are negative
+      if(size != values.size && maxValue < 0){
+        maxIdx = calcInactiveIdx(indices(0))
+        maxValue = 0
+      }
+      maxIdx
+    }
+  }
+
+  /**
+   * Calculates the first instance of an inactive node in a sparse vector and returns the Idx
+   * of the element.
+   * @param idx starting index of computation
+   * @return index of first inactive node or -1 if it cannot find one
+   */
+  private[SparseVector] def calcInactiveIdx(idx: Int): Int ={
```

Comment:
I think the way I had it previously is a little bit cleaner from a scala perspective, though I am still getting used to the language. Trying with a loop and then if checks like above is forcing me to keep track of some extra over head variables. I cant seem just do the below without having the rest of the else statement return something as well. This could just be lack of knowledge on my part about the scala langauge 
```
if (isAwesome){
  return true
}
```